kernel/dev/serial/rs232.cpp

```diff
@@ -43,17 +45,44 @@ static char px_rs232_read_char() {
     return px_read_byte(rs_232_port_base + RS_232_DATA_REG);
 }
 
-static void px_rs232_write_char(char c) {
-    px_mutex_lock(&mutex_rs232);
+static inline void px_rs232_write_char(char c) {
     while (px_rs232_is_transmit_empty() == 0);
     px_write_byte(rs_232_port_base + RS_232_DATA_REG, c);
-    px_mutex_unlock(&mutex_rs232);
+}
+
+static int vprintf_helper(unsigned c, void **ptr)
+{
+    (void) ptr;
+    px_rs232_write_char((char)c);
+    return 0;
+}
+
+int px_rs232_vprintf(const char* fmt, va_list args)
+{
+    int retval;
+    //px_mutex_lock(&mutex_rs232);
```

Comment:
Shouldn't these locks be removed if they're commented out?

Reply:
Probably, but I left them in because I'd like to have some locking for regular tasks there, but to not have locking for scheduling debug code.

Reply:
You could just do an `#ifndef DEBUG`

Reply:
But this function can be used in both places in the same build

Reply:
I'll just remove the comments for now